base/libgit2/tag.jl

```diff
@@ -35,6 +35,7 @@ function name(tag::GitTag)
 end
 
 function target(tag::GitTag)
-    oid_ptr = Ref(ccall((:git_tag_target_id, :libgit2), Ptr{Oid}, (Ptr{Void}, ), tag.ptr))
-    return oid_ptr[]
+    oid_ptr = ccall((:git_tag_target_id, :libgit2), Ptr{Oid}, (Ptr{Void}, ), tag.ptr)
+    oid_ptr == C_NULL && return Oid()
```

Comment:
A zero default returning from this function means you won't see the problem until later, which makes it more difficult to find out what went wrong. Throw errors as early as possible. If it's not possible to throw them before this function, then throw them here.